### PR TITLE
[ISSUE-293][REFACTOR] Init worker rpc endpoint and reserve slot in parallel to speed up register shuffle process

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

File: client/src/main/scala/com/aliyun/emr/rss/client/write/LifecycleManager.scala [hunk 1]
```diff
@@ -294,10 +294,11 @@ class LifecycleManager(appId: String, val conf: RssConf) extends RpcEndpoint wit
     // won't be empty since master will reply SlotNotAvailable status when reserved slots is empty.
     val slots = res.workerResource
     val candidatesWorkers = new util.HashSet(slots.keySet())
-    val connectFailedWorkers = new util.ArrayList[WorkerInfo]()
+    val connectFailedWorkers = ConcurrentHashMap.newKeySet[WorkerInfo]()
 
     // Second, for each worker, try to initialize the endpoint.
-    slots.asScala.foreach { case (workerInfo, _) =>
+    val parallelism = Math.min(slots.size(), RssConf.rpcMaxParallelism(conf))
+    ThreadUtils.parmap(slots.asScala.to, "CommitFiles", parallelism) { case (workerInfo, _) =>
```
Comment (FMX) [inline client/src/main/scala/com/aliyun/emr/rss/client/write/LifecycleManager.scala:301]: The prefix should be changed.

Comment (AngersZhuuuu) [reply client/src/main/scala/com/aliyun/emr/rss/client/write/LifecycleManager.scala]: DOne

**Comment by FMX** [comment]
LGTM.

**Pull request merged**